### ISSUE TITLE
OnClick event triggered by other inputs.

### ISSUE THREAD

**Issue by vrtulka23** [blackary/streamlit-image-coordinates#17]
I have something like this:

`cols = st.columns([2,1])
    draw = ImageDraw.Draw(image2)
    draw.ellipse((20, 20, 180, 180), fill = None, outline ='blue')
    with cols[0]:
        value = streamlit_image_coordinates(image2)
        if value:
            st.session_state['points'].append(value)
    cols[1].number_input("False positive",format="%i",step=1,key=f"img_proc_pos_{i:d}")
    cols[1].number_input("Misses",format="%i",step=1,key=f"img_proc_neg_{i:d}")
    cols[1].button("Clean points", on_click=clean_points)`

If I click on any of the inputs or button above, I will trigger an onclick event on the image and streamlit_image_coordinates will return a value of the last clicked point on the image.

**Comment by vrtulka23** [comment]
Just to clarify: clicking on other components, like inputs and buttons, should lead to last "value", but to a None "value".

**Comment by vrtulka23** [comment]
Yes, I observe that when other components (e.g. inputs,...) are clicked on, the streamlit_image_coordinates returns the previously clicked coordinates... So, if I am recording the coordinates into a session (e.g. with a condition "if value:"), every click of another component leads to a false coordinate record... 
So far I could solve this by saving the last coordinate into a session and than comparing if the new "image click" value is the same, or not.

**Comment by blackary** [comment]
This is a fundamental Streamlit component behavior: when any input triggers a script re-run, Streamlit returns the last cached component value, not `None`. The component itself only calls `setComponentValue` on actual clicks, but the Python side always returns the previously-set value on re-render.

The cleanest solution is to use the `on_click` callback (available since v0.4.0), which fires only when the component genuinely sends a new value — not on re-runs:

```python
def record_point():
    coord = st.session_state["img"]
    st.session_state.points.append(coord)

streamlit_image_coordinates(image, key="img", on_click=record_point)
```

If you need to use the return value directly, comparing `unix_time` (which the component also returns) against a stored last-click timestamp is the reliable approach — as you've already discovered.
